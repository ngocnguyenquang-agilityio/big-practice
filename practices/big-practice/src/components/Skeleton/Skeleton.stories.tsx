// Types
import type { Meta, StoryObj } from '@storybook/react';

// Components
import Skeleton from '.';

const meta = {
  title: 'Components/Skeleton',
  component: Skeleton,
} satisfies Meta<typeof Skeleton>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    numOfItems: 1,
  },
};
