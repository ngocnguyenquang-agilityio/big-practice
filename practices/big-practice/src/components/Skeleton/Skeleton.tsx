const Skeleton = ({ pagination = 1 }: { pagination?: number }) => {
  const listItems = [];

  for (let i = 1; i <= pagination; i++) {
    listItems.push(
      <li
        className='aspect-square transition-opacity animate-fadeIn'
        key={i}
      >
        <div className='relative inline-block h-full w-full animate-pulse'>
          <div className='group flex h-full w-full items-center justify-center overflow-hidden rounded-lg border hover:border-blue-600 bg-black relative border-neutral-800'>
            <div className='flex items-center justify-center w-full h-full rounded sm:w-96 bg-gray-700'>
              <svg
                className='w-10 h-10 text-gray-600'
                aria-hidden='true'
                xmlns='http://www.w3.org/2000/svg'
                fill='currentColor'
                viewBox='0 0 20 18'
              >
                <path d='M18 0H2a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2Zm-5.5 4a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3Zm4.376 10.481A1 1 0 0 1 16 15H4a1 1 0 0 1-.895-1.447l3.5-7A1 1 0 0 1 7.468 6a.965.965 0 0 1 .9.5l2.775 4.757 1.546-1.887a1 1 0 0 1 1.618.1l2.541 4a1 1 0 0 1 .028 1.011Z' />
              </svg>
            </div>
          </div>
        </div>
      </li>,
    );
  }

  return (
    <ul
      className='grid grid-flow-row gap-4 grid-cols-1 sm:grid-cols-2 lg:grid-cols-3'
      aria-label='skeleton'
    >
      {listItems}
    </ul>
  );
};

export default Skeleton;
