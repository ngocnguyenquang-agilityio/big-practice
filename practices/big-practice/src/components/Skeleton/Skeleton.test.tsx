import { RenderResult, render, within } from '@testing-library/react';

// Components
import Skeleton from '.';

const props = {
  pagination: 2,
};

describe('Skeleton Component', () => {
  let comp: RenderResult;
  beforeEach(() => {
    comp = render(<Skeleton {...props} />);
  });

  test('Should render successfully', () => {
    expect(comp).toBeTruthy();
  });

  test('Should match snapshot', () => {
    expect(comp).toMatchSnapshot();
  });

  test('Should render correct number of items', () => {
    const list = comp.getByRole('list', {
      name: /skeleton/i,
    });
    const { getAllByRole } = within(list);
    const items = getAllByRole('listitem');
    expect(items.length).toBe(2);
  });
});
