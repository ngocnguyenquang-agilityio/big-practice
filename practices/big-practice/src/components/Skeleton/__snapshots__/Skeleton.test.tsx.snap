// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton Component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        aria-label="skeleton"
        class="grid grid-flow-row gap-4 grid-cols-1 sm:grid-cols-2 lg:grid-cols-3"
      >
        <li
          class="aspect-square transition-opacity animate-fadeIn"
        >
          <div
            class="relative inline-block h-full w-full animate-pulse"
          >
            <div
              class="group flex h-full w-full items-center justify-center overflow-hidden rounded-lg border hover:border-blue-600 bg-black relative border-neutral-800"
            >
              <div
                class="flex items-center justify-center w-full h-full rounded sm:w-96 bg-gray-700"
              >
                <svg
                  aria-hidden="true"
                  class="w-10 h-10 text-gray-600"
                  fill="currentColor"
                  viewBox="0 0 20 18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M18 0H2a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2Zm-5.5 4a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3Zm4.376 10.481A1 1 0 0 1 16 15H4a1 1 0 0 1-.895-1.447l3.5-7A1 1 0 0 1 7.468 6a.965.965 0 0 1 .9.5l2.775 4.757 1.546-1.887a1 1 0 0 1 1.618.1l2.541 4a1 1 0 0 1 .028 1.011Z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </li>
        <li
          class="aspect-square transition-opacity animate-fadeIn"
        >
          <div
            class="relative inline-block h-full w-full animate-pulse"
          >
            <div
              class="group flex h-full w-full items-center justify-center overflow-hidden rounded-lg border hover:border-blue-600 bg-black relative border-neutral-800"
            >
              <div
                class="flex items-center justify-center w-full h-full rounded sm:w-96 bg-gray-700"
              >
                <svg
                  aria-hidden="true"
                  class="w-10 h-10 text-gray-600"
                  fill="currentColor"
                  viewBox="0 0 20 18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M18 0H2a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2Zm-5.5 4a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3Zm4.376 10.481A1 1 0 0 1 16 15H4a1 1 0 0 1-.895-1.447l3.5-7A1 1 0 0 1 7.468 6a.965.965 0 0 1 .9.5l2.775 4.757 1.546-1.887a1 1 0 0 1 1.618.1l2.541 4a1 1 0 0 1 .028 1.011Z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      aria-label="skeleton"
      class="grid grid-flow-row gap-4 grid-cols-1 sm:grid-cols-2 lg:grid-cols-3"
    >
      <li
        class="aspect-square transition-opacity animate-fadeIn"
      >
        <div
          class="relative inline-block h-full w-full animate-pulse"
        >
          <div
            class="group flex h-full w-full items-center justify-center overflow-hidden rounded-lg border hover:border-blue-600 bg-black relative border-neutral-800"
          >
            <div
              class="flex items-center justify-center w-full h-full rounded sm:w-96 bg-gray-700"
            >
              <svg
                aria-hidden="true"
                class="w-10 h-10 text-gray-600"
                fill="currentColor"
                viewBox="0 0 20 18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 0H2a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2Zm-5.5 4a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3Zm4.376 10.481A1 1 0 0 1 16 15H4a1 1 0 0 1-.895-1.447l3.5-7A1 1 0 0 1 7.468 6a.965.965 0 0 1 .9.5l2.775 4.757 1.546-1.887a1 1 0 0 1 1.618.1l2.541 4a1 1 0 0 1 .028 1.011Z"
                />
              </svg>
            </div>
          </div>
        </div>
      </li>
      <li
        class="aspect-square transition-opacity animate-fadeIn"
      >
        <div
          class="relative inline-block h-full w-full animate-pulse"
        >
          <div
            class="group flex h-full w-full items-center justify-center overflow-hidden rounded-lg border hover:border-blue-600 bg-black relative border-neutral-800"
          >
            <div
              class="flex items-center justify-center w-full h-full rounded sm:w-96 bg-gray-700"
            >
              <svg
                aria-hidden="true"
                class="w-10 h-10 text-gray-600"
                fill="currentColor"
                viewBox="0 0 20 18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 0H2a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2Zm-5.5 4a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3Zm4.376 10.481A1 1 0 0 1 16 15H4a1 1 0 0 1-.895-1.447l3.5-7A1 1 0 0 1 7.468 6a.965.965 0 0 1 .9.5l2.775 4.757 1.546-1.887a1 1 0 0 1 1.618.1l2.541 4a1 1 0 0 1 .028 1.011Z"
                />
              </svg>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
